### DISCUSSION TITLE
Asthma transition based on air pollution

### DISCUSSION THREAD

**Discussion by schmittcp** [synthetichealth/synthea#1240] [Q&A]
Hi,

I am involved in a project where it would be very beneficial to have an asthma model that incorporated exposure to air pollution.   Naively, I would think the best way is to allow a transition that queries external software with the date and location of the patient and the software returns the probability of the transition especially as we would be interested in simulating different exposure impact models.  Is this feasible using this type of approach, or might there be a separate type of approach we could consider?   We are happy to do some development to support this, but would appreciate guidance on how to approach it.

Regards,

Charles

**Comment by jawalonoski** [comment]
I think it depends on the resolution of your air pollution data... for instance, if it was coarse enough to put in a flat file (e.g., maybe by year and zip code) you might consider using a Table Transition ([wiki](https://github.com/synthetichealth/synthea/wiki/Generic-Module-Framework%3A-Transitions#table), [code](https://github.com/synthetichealth/synthea/blob/master/src/main/java/org/mitre/synthea/engine/Transition.java#L216)).

On the other hand, if it was very fine resolution (e.g., weekly data by lat/lon) or dynamic, you might consider building a new transition type that queries a database or API or something. In which case, you need to implement a new class that extends `Transition` and implements the `public abstract String follow(Person person, long time)` method. ([code](https://github.com/synthetichealth/synthea/blob/master/src/main/java/org/mitre/synthea/engine/Transition.java#L28))

If you go the custom-transition route, you'll need to modify your module JSON by hand to use that transition-type, unless you also modify the Module Builder to support that transition-type.

There are other ways to solve this problem, for instance, you could make a hard-coded Java module that queries the database every time step and sets the probability within an attribute on the Patient... and later the asthma module just uses that attribute value as the distributed transition probability. See "named probabilities" ([wiki](https://github.com/synthetichealth/synthea/wiki/Generic-Module-Framework%3A-Transitions#named-probabilities))

**Reply by schmittcp** [reply]
Thanks for the reply.  We would generally want fine resolution (daily data by lat/lon).  Both approaches sound reasonable and doable, so its great to hear the system was designed such that this need has a path forward.  I think we'll spend a little time better understanding the architecture and how much flexibility we would want before deciding on which approach to take (although extending Transition sounds like the more flexible approach).
